src/main/java/com/diffblue/demo/ecommerce/controllers/CheckoutController.java

```diff
@@ -87,4 +109,78 @@ public String viewPaymentPage(Map<String, Object> model, HttpSession session) {
     }
   }
 
+  /**
+   * Validate payment form and make payment.
+   * @param paymentForm - payment form object
+   * @param bindingResult - object with form validation result
+   * @param model - model where to add data to access on view
+   * @param session - current http session
+   * @return page for the output
+   */
+  @PostMapping("/payment")
+  public String makePayment(@Valid PaymentForm paymentForm, BindingResult bindingResult,
```

Comment:
Please, could you add comments to the method? It's almost worth over commenting based on us using this as a demo.